### ISSUE TITLE
Cannot add Nextcloud instance

### ISSUE THREAD

**Issue by TylerLaBree** [alainm23/planify#1385]
**Describe the bug**

Recently, I haven't been able to sync with my home network Nextcloud server. I tried clearing app data, and setting the connection to the server up again. When I add the server (port 443), I get the following error:

```
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>405 Method Not Allowed</title>
</head><body>
<h1>Method Not Allowed</h1>
<p>The requested method PROPFIND is not allowed for this URL.</p>
<hr>
<address>Apache/2.4.61 (Debian) Server at 192.168.1.131 Port 80</address>
</body></html>
```

Note that changing the port to an incorrect one provides the same error. 

**To Reproduce**

1. Go to Main Menu->Preferences->Integrations->Add Source->Nextcloud
2. Fill information, with Server URL using http and port 443
3. Select Log In

**Expected behavior**

Planify should connect to the server and sync files.

**Screenshots**

![image](https://github.com/user-attachments/assets/39c0aa51-c7f0-426e-96f3-20dfc92f48b3)

![image](https://github.com/user-attachments/assets/afd21765-4385-477b-adfa-07a735c68c11)

**Desktop:**
 - OS: Fedora Linux 40
 - DE: Gnome 46.3.1

**Additional context**

Syncing worked ~2 weeks ago. Connecting to my Nextcloud instance via my web browser still works. Gnome Calendar still connects and syncs through Gnome Online Accounts->Nextcloud. DAVx5 still works for syncing my calendar and tasks to my phone.

It looks like it's having an issue with port 80. Possibly either Planify is not respecting the port I tell it, or it needs to connect to port 80 before port 443. It could be something wrong with my Nextcloud configuration, although I don't believe I've made any changes to it in this time.

Since running into the issue, I've updated my Nextcloud server, my server packages, and rebooted everything. The error remains the same.

Visiting port 80 in a web browser takes me to the Apache2 Debian Default Page.

**Comment by alainm23** [comment]
@TylerLaBree  Hello and thanks for reporting this issue. For now, Planify does not support ports other than 80 (I have not tested connecting with IP and port directly), maybe it is too much to ask but could you lend me a test account.

**Comment by TylerLaBree** [comment]
Hi @alainm23, I can't share an account with you since the instance is only accessible on my home network, and I'm not comfortable enough with my security knowledge to set up port forwarding.

Is the requirement to have the server on port 80 recently added? Is there a way to work around this requirement?

I plan to get a DNS server set up eventually so I don't have to connect to ip:port, and instead to something like cloud.home.arpa. Would this fix my issue?

Thanks for your work on this great project!

**Comment by alainm23** [comment]
Don't worry, I haven't pllged to test these cases, Planify uses a proprietary service to sync with CalDAV, I will install an own instance locally and give you a solution, thank you very much.

**Comment by jkgagnon** [comment]
@TylerLaBree .  I had this same issue as well after updating past 4.10. I am running nextcloud as an app on truenas scale, and synced solely from my LAN. I solved by installing an older branch of Planify. The noted improvement: "better nextcloud flow" was not better for me, and does not prompt for IP address plus username and password.

As a workaround, I installed an older branch, setup Nextcloud on it again, and then upgraded to the newest version. It worked for me twice.

`flatpak remote-info --log flathub io.github.alainm23.planify`

I chose a branch id that worked:

`sudo flatpak update --commit=7c93c829ba74cf95607e69a980a7585695bf26bcbe9185a4fcb62544689504b0 io.github.alainm23.planify`

You'll get this if it works:
`Looking for updates…


        ID                             Branch     Op    Remote     Download
 1. [✓] io.github.alainm23.planify     stable     u     flathub    10.9 MB / 12.3 MB

Updates complete.
`
It's not ideal, and you will then need to run another command to force it off of the selected branch id, which I believe freezes the version at that outdated one.
